controllers/ibmpowervscluster_controller_test.go

```diff
@@ -294,24 +309,318 @@ func TestIBMPowerVSClusterReconciler_Reconcile(t *testing.T) {
 func TestIBMPowerVSClusterReconciler_reconcile(t *testing.T) {
 	testCases := []struct {
 		name                string
-		powervsClusterScope *scope.PowerVSClusterScope
+		powervsClusterScope func() *scope.PowerVSClusterScope
 		clusterStatus       bool
+		expectedResult      ctrl.Result
+		expectedError       error
 	}{
 		{
 			name: "Should add finalizer and reconcile IBMPowerVSCluster",
-			powervsClusterScope: &scope.PowerVSClusterScope{
-				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{},
+			powervsClusterScope: func() *scope.PowerVSClusterScope {
+				return &scope.PowerVSClusterScope{
+					IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{},
+				}
 			},
-			clusterStatus: false,
 		},
 		{
 			name: "Should reconcile IBMPowerVSCluster status as Ready",
-			powervsClusterScope: &scope.PowerVSClusterScope{
-				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
-					ObjectMeta: metav1.ObjectMeta{
-						Finalizers: []string{infrav1beta2.IBMPowerVSClusterFinalizer},
+			powervsClusterScope: func() *scope.PowerVSClusterScope {
+				return &scope.PowerVSClusterScope{
+					IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+						ObjectMeta: metav1.ObjectMeta{
+							Finalizers: []string{infrav1beta2.IBMPowerVSClusterFinalizer},
+						},
 					},
-				},
+				}
+			},
+			clusterStatus: true,
+		},
+		{
+			name: "When Power VS zone does not support PER",
+			powervsClusterScope: func() *scope.PowerVSClusterScope {
+				return &scope.PowerVSClusterScope{
+					IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+						ObjectMeta: metav1.ObjectMeta{
+							Finalizers:  []string{infrav1beta2.IBMPowerVSClusterFinalizer},
+							Annotations: map[string]string{infrav1beta2.CreateInfrastructureAnnotation: "true"},
+						},
+					},
+				}
+			},
+			expectedError: errors.New("PowerVS zone is not set"),
+		},
+		{
+			name: "When resource group name is not set",
+			powervsClusterScope: func() *scope.PowerVSClusterScope {
+				clusterScope := &scope.PowerVSClusterScope{
+					IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+						ObjectMeta: metav1.ObjectMeta{
+							Finalizers:  []string{infrav1beta2.IBMPowerVSClusterFinalizer},
+							Annotations: map[string]string{infrav1beta2.CreateInfrastructureAnnotation: "true"},
+						},
+						Spec: infrav1beta2.IBMPowerVSClusterSpec{
+							Zone: ptr.To("dal10"),
+						},
+					},
+				}
+				mockPowerVS := powervsmock.NewMockPowerVS(gomock.NewController(t))
+				mockPowerVS.EXPECT().GetDatacenterCapabilities(gomock.Any()).Return(map[string]bool{"power-edge-router": true}, nil)
+				clusterScope.IBMPowerVSClient = mockPowerVS
+				return clusterScope
+			},
+			expectedError: errors.New("resource group name is not set"),
+		},
+		{
+			name: "When reconcile Power VS resource returns requeue",
```

Comment:
```suggestion
			name: "When reconcile Power VS resource returns requeue as true",
```
Please change wherever applicable